javaharness/java/arcs/api/ShellApi.java

```diff
@@ -1,7 +1,27 @@
 package arcs.api;
 
-/** An API for Java to call Javascript (shell). */
-public interface ShellApi {
-  // Sends a message to Arcs JS shell.
-  String receive(String json);
-}
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class ShellApi {
+
+  public interface Proxy {
+    void onMessage(String message);
```

Comment:
I feel like this should be called sendMessage, it sounds like an event handler if named onXXX.

Reply:
Renamed.